rest-dashboard.go

```diff
@@ -57,6 +57,46 @@ type BoardProperties struct {
 	FolderURL   string    `json:"folderUrl"`
 }
 
+//RawBoardRequest struct that wraps Board and parameters being sent
+type RawBoardRequest struct {
+	Dashboard  []byte
+	Parameters SetDashboardParams
+}
+
+//MarshalJSON serializes the request to match the expectations of the grafana API.
+// Additionally, if preseveID is false, then the dashboard id is set to 0
+func (d RawBoardRequest) MarshalJSON() ([]byte, error) {
+	buf := bytes.NewBufferString("")
+	var raw []byte
+
+	if d.Parameters.PreserveId {
+		raw = d.Dashboard
+	} else {
+		var plain map[string]interface{} = make(map[string]interface{})
+		if err := json.Unmarshal(d.Dashboard, &plain); err != nil {
+			return nil, err
+		}
+		plain["id"] = 0
+		raw, _ = json.Marshal(plain)
+
+	}
+
+	parameters, err := json.Marshal(d.Parameters)
```

Comment:
The code below seems a bit hacky. Why not:
```
  var output struct {
    Dashboard []byte
    SetDashboardParams
  }
  
  output.Dashboard = raw
  output.SetDashboardParams = parameters
  
  return json.Marshal(output)
```
:thinking: ? It's much simpler and should do the same. (https://play.golang.org/p/nssZ3HPJOjG)

Reply:
yeah that's way cleaner.  I was trying to figure out a way to do that without string manipulation.  Updated.

Reply:
This is now resolved!